x-pack/plugins/security_solution/public/overview/containers/overview_risky_host_links/use_risky_hosts_dashboard_links.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useState, useEffect } from 'react';
+import { SavedObjectAttributes } from '@kbn/securitysolution-io-ts-alerting-types';
+import { useKibana } from '../../../common/lib/kibana';
+import { LinkPanelListItem } from '../../components/link_panel';
+
+const DASHBOARD_REQUEST_BODY_SEARCH = '"Drilldown of Host Risk Score"';
+export const DASHBOARD_REQUEST_BODY = {
+  type: 'dashboard',
+  search: DASHBOARD_REQUEST_BODY_SEARCH,
+  fields: ['title'],
+};
+
+export const useRiskyHostsDashboardLinks = (
+  to: string,
+  from: string,
+  listItems: LinkPanelListItem[]
+) => {
+  const createDashboardUrl = useKibana().services.dashboard?.dashboardUrlGenerator?.createUrl;
+  const savedObjectsClient = useKibana().services.savedObjects.client;
+
+  const [listItemsWithLinks, setListItemsWithLinks] = useState(listItems);
```

Comment:
I think we have a bug here. You can see in the video, that the list of links in the panel doesn't reflect on time range changes.

Looks like it happens, because, in this hook, we use `listItems` as the initial value for `listItemsWithLinks`.

After time range change in `listItems` can be a different value, but `listItemsWithLinks` doesn't reflect and has an old one.

https://user-images.githubusercontent.com/7609147/132328617-80a5c7a1-8018-4a0f-ae4b-5c5548689a59.mov

Reply:
Also, we have a similar bug, when Dashboard is enabled. 
In the video, you can see how when I change the time range to **today**, 
- it updates the panel and shows empty panel for 1 sec (correct behaviour)
- and then again show a list of links (bug)

Looks like it happens because in useEffect we have some kind of async race condition.
Let's say we have 3 elements in `listItems`
When we change the time range whats happens

process A. 
we go to useEffect (I assume because _from_ or _to_ is changed). **listItems** has 3 elements, we initiate 3 requests and waiting for them to finish

process B. 
In the meantime, we again go to useEffect, because **listItems** is changed and now has 0 elements. 
we call **setListItemsWithLinks([])** and now **listItemsWithLinks == []**

process A. 
After some time the 3 requests are finished, and we call **setListItemsWithLinks([3 elements])**  and now **listItemsWithLinks** has 3 elements


https://user-images.githubusercontent.com/7609147/132337330-708ce689-1af1-4ce6-b2da-738c63caa171.mov

Reply:
fixed with an "effect cancellation" similar to 

```
useEffect(() => {
    let canceled = false;
    fetch(id).then(data => {
      if (!canceled) {
        setData(data);
      }
    });
    return () => (canceled = true);
  }, [id]);
```

thank you so much for identifying the bug & solution!